engine/src/cmdss.cpp

```diff
@@ -448,16 +448,22 @@ Parse_stat MCGo::parse(MCScriptPoint &sp)
 MCStack *MCGo::findstack(MCExecContext &ctxt, MCStringRef p_value, Chunk_term etype, MCCard *&cptr)
 {
     MCStack *sptr = nil;
+    sptr = MCInterfaceTryToEvalStackFromString(p_value);
+    if (sptr != nil)
+        return sptr;
+
     if (MCInterfaceStringCouldBeStack(p_value))
     {
-        sptr = MCInterfaceTryToEvalStackFromString(p_value);
-        if (sptr == nil)
-        {
+        //sptr = MCInterfaceTryToEvalStackFromString(p_value);
+        //if (sptr == nil)
+        //{
             if (MCresult->isclear())
                 ctxt . SetTheResultToCString("can't build stack from string");
-        }
+        //}
         return sptr;
     }
+    else
+        MCresult->clear();
     
 	if (etype == CT_STACK)
         return NULL;
```

Comment:
That's a relic of old code. For a while we started to use `nil` by preference over `NULL`. Now, in new code, `nullptr` is preferred. Rather than do any kind of global replace (which is a bit of a nightmare for merging and so on), we just change them when the relevant code is touched.